### ISSUE TITLE
Swift Compiler Optimization Level is left at "Fast" for custom debug build configurations

### ISSUE THREAD

**Issue by tumatauenga** [CocoaPods/CocoaPods#5122]
* [ 🌈  ] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

## What did you do?

1. Created a project with a few additional "debug" build configurations for the purpose of pointing to different environments during development
2. Listed these configurations using the `xcodeproj` command along with the `=> :debug` modifier to indicate that they were debug configurations
3. ran `pod install`

## What did you expected to happen?

- I expected that in the Pods project, the optimization level for the swift compiler would be set to "None" for the build configurations marked as `:debug`

## What happened instead?

- The swift compiler optimization level was set to "None" for the "Debug" build configuration, but was left at "Fast" for my custom debug build configurations. This made debugging difficult when the code was run since any code in the Pods project was optimized, and also meant that `assert` statements were ignored. This was particularly problematic given that I'm currently developing a custom private CocoaPod to manage code shared across a few different projects.

## Podfile

```ruby
`xcodeproj './MyProject.xcodeproj', 'DebugDevelopment' => :debug, 'DebugQA' => :debug, 'DebugDR' => :debug, 'DebugProduction' => :debug, 'HockeyApp' => :release, 'TestFlight' => :release, 'AppStore' => :release`
```

**Comment by segiddins** [comment]
This should be fixed as of the 1.0 betas.

**Comment by neonichu** [comment]
This issue has been inactive for a long time and will be closed because we want to move forward with a clean slate after our 1.0 release. If you believe this is still relevant, please retest with the 1.0 release candidate and comment with a reproducible project in order for this to become actionable again. Thanks!
